### PR TITLE
provide a default connect timeout for http client, closes #1230

### LINKED ISSUE #1230: Provide configurable default connect timeout for http client
Currently only read timeout is configurable on http client. It will be useful to configure the connect timeout to values other than 30 seconds. For example when using retrofit, we are not able to specify timeouts per request.

### PR DESCRIPTION
**Pull request by drmaas**
This change is [Reviewable](https://reviewable.io/reviews/ratpack/ratpack/1239)

### CODE REVIEW

**Comment by drmaas** [comment]
@beckje01 opened a new PR, but the build is still missing.

**Comment by beckje01** [comment]
Yeah I see an odd message in CI I'll look into it

**Comment by drmaas** [comment]
@beckje01 ratpack-manual tests are failing ... can we merge this as it is unrelated to those?

```
> Task :ratpack-manual:test

ratpack.manual.ManualCodeSnippetTests > 13-http.md:213 FAILED
    org.junit.ComparisonFailure: expected:<[Sent using render(Object)!]> but was:<[<!doctype html><title>404 Not Found</title><h1 style="text-align: center">404 Not Found</h1><hr/><p style="tex
253 tests completed, 1 failed

```

**Comment by drmaas** [comment]
ping

File: ratpack-core/src/main/java/ratpack/http/client/HttpClientSpec.java [hunk 1]
```diff
@@ -77,15 +77,25 @@
...
    * <p>
-   * Increasing this value can increase throughput at the expense of memory use.
+   * Defaults to 30 seconds.
    *
-   * @param numBytes the max number of bytes to emit
+   * @param connectTimeout the read timeout value for responses
    * @return {@code this}
-   * @since 1.5
    */
+  HttpClientSpec connectTimeout(Duration connectTimeout);
```
Comment (beckje01) [inline ratpack-core/src/main/java/ratpack/http/client/HttpClientSpec.java]: Needs the `@since 1.5`

**Comment by beckje01** [comment]
Overall it looks good the error in the tests will be fine when we merge.

**Comment by drmaas** [comment]
Great, thanks.

**Pull request merged**